# 1.10 Olfactory receptor detail page

## Introduction

<p style="text-align:justify; line-height: 1.8; ">In CORD, each olfactory receptor is equipped with a dedicated details page designed to provide comprehensive information about the receptor from various perspectives.</p>

## Basic Information

<div style="text-align:center;">
    <img src="../data/2-figures/1-Guides\1.10.1_basic.webp" alt="1.10.1_basic" needTransformUrl="true" />
</div>
<p style="text-align:justify; line-height: 1.8; ">Presenting the crucial fundamental information of the receptor, chiefly comprising the following details:</p>

* <p style="text-align:justify; line-height: 1.8; ">Entry: The unique identifier for each olfactory receptor in this database.</p>
* <p style="text-align:justify; line-height: 1.8; ">Match Score: Scoring of the number of matched patterns. For details, please <a href="#/./2-DataIntroduction/2.20-PatternMatch.md">refer to here</a>.</p>
* <p style="text-align:justify; line-height: 1.8; ">Gene Name: Identifier for gene level in our classification system.</p>
* <p style="text-align:justify; line-height: 1.8; ">Pattern Match: If the pattern is matched in the sequence, the corresponding square is green, otherwise it is red. For details, please <a href="#/./2-DataIntroduction/2.20-PatternMatch.md">refer to here</a>.</p>
* <p style="text-align:justify; line-height: 1.8; ">Gene Family: Identifier for gene family level in our classification system.</p>
* <p style="text-align:justify; line-height: 1.8; ">Gene Subfamily: Identifier for gene subfamily level in our classification system.</p>
* <p style="text-align:justify; line-height: 1.8; ">Functional/Pseudo: Indication of whether this entry is a functional olfactory receptor gene or a pseudogene.</p>
* <p style="text-align:justify; line-height: 1.8; ">External Name: The gene name as documented in the UniProt database.</p>
* <p style="text-align:justify; line-height: 1.8; ">Amino Acids: The count of amino acids in the protein sequence of this olfactory receptor entry.</p>
* <p style="text-align:justify; line-height: 1.8; ">Annotation Date: The annotation date of this entry in the database.</p>
* <p style="text-align:justify; line-height: 1.8; ">NCBI Taxonomy: The unique identifier assigned by the NCBI to the source organism.</p>
* <p style="text-align:justify; line-height: 1.8; ">Organism: The unique name for an organism such that two people can be sure that they are referring to the same organism.</p>
* <p style="text-align:justify; line-height: 1.8; ">Common Name: The species common name.</p>

## Sequences

<div style="text-align:center;">
    <img src="../data/2-figures/1-Guides/1.10.2_sequence.webp" alt="1.10.2_sequence" needTransformUrl="true" />
</div>
<p style="text-align:justify; line-height: 1.8; ">This section comprises a sequence display module designed specifically for showcasing the protein and DNA sequences associated with the receptor.</p>


<p style="text-align:justify; line-height: 1.8; "><span style="font-size:18px; color:red">①</span> Color Scheme Radio Button</p>

* Chemistry: Coloring based on the chemical properties of amino acids.

| <div style="width:11em">Propertie</div> | <div style="width:20em">Amino acid</div> | <div style="width:10em">Color</div> |
| --------------------------------------- | ---------------------------------------- | ----------------------------------- |
| Polar                                   | G, S, T, Y, C                            | <font color=green>green</font>      |
| Neutral                                 | Q, N                                     | <font color=purple>purple</font>    |
| Basic                                   | K, R, H                                  | <font color=blue>blue</font>        |
| Acidic                                  | D, E                                     | <font color=red>red</font>          |
| Hydrophobic                             | A, V, L, I, P, W, F, M                   | <font color=black>black</font>      |

* Charge: Coloring based on the charge properties of amino acids.

| <div style="width:11em">Propertie</div> | <div style="width:20em">Amino acid</div> | <div style="width:10em">Color</div> |
| --------------------------------------- | ---------------------------------------- | ----------------------------------- |
| Positive                                | K, R, H                                  | <font color=blue>blue</font>        |
| Negative                                | D, E                                     | <font color=red>red</font>          |

* Hydrophobicity: Coloring based on the hydrophobic properties of amino acids.

| <div style="width:11em">Propertie</div> | <div style="width:20em">Amino acid</div> | <div style="width:10em">Color</div> |
| --------------------------------------- | ---------------------------------------- | ----------------------------------- |
| Hydrophilic                             | R, K, D, E, N, Q                         | <font color=blue>blue</font>        |
| Neutral                                 | S, G, H, T, A, P                         | <font color=green>green</font>      |
| Hydrophobic                             | Y, V, M, C, L, F, I, W                   | <font color=black>black</font>      |

* Clear: Clear the color scheme.

<span style="font-size:18px; color:red">②</span> Download sequence button.

<p style="text-align:justify; line-height: 1.8; "><span style="font-size:18px; color:red">③</span> Shortcut button to navigate to the BLAST page. This function will automatically import the sequence from the current page into the input box of BLAST.</p>

## Snake Diagram

<div style="text-align:center;">
    <img src="../data/2-figures/1-Guides/1.10.3_snake.webp" alt="1.10.3_snake" needTransformUrl="true" />
</div>
<p style="text-align:justify; line-height: 1.8; ">Snake diagram showing the main structural features of a olfactory receptor. The protein single-chain spans the cytoplasmic membrane via seven transmembrane helices (TM1-7), and it is characterized by three ECLs (ECL1-3) and three ICLs (ICL1-3), as well as a helical intracellular C terminus. For details, please <a href="#/./2-DataIntroduction/2.18-SnakeDiagram.md">refer to here</a>.</p>

<p style="text-align:justify; line-height: 1.8; "><span style="font-size:18px; color:red">①</span> Color Scheme Radio Button. The color schemes for Chemistry, Charge, and Hydrophobicity are identical to those of the Snake diagram.</p>

* Coloring of the circles in the snake diagram is based on the properties of amino acids. There are three color schemes to choose from: Chemistry, Charge, and Hydrophobicity. The details of these schemes are identical to those outlined in the above "Sequence" 
* Binding pockets:  The predicted odor molecules bind to the pocket. In comparison to the previous version, this time we have incorporated structural information from OR51E2.
* OR .50: The residues at the OR generic residue numbering "X.50" in each transmembrane helical region.
* Class A GPCR .50: The residues at the class A GPCR generic residue numbering "X.50" in each transmembrane helical region.
* Clear: Remove the color scheme.

<span style="font-size:18px; color:red">②</span> Download image button.

## Structure

<div style="text-align:center;">
    <img src="../data/2-figures/1-Guides/1.10.4_structure.webp" alt="1.10.4_structure" needTransformUrl="true" />
</div>

Protein structure module. For detailed instructions, please [refer to here](https://www.youtube.com/watch?v=bV3z4wWp-Ks).

<div style="text-align:center;">
    <img src="../data/2-figures/1-Guides/1.10.4_structure_quality_1.webp" alt="1.10.4_structure_quality_1" needTransformUrl="true" />
</div>
<p style="text-align:justify; line-height: 1.8; ">Protein structure module. Linear plot of the pLDDT scores for each residue of the receptor. The horizontal axis represents the amino acid positions, while the vertical axis represents the pLDDT values.</p>

<div style="text-align:center;">
    <img src="../data/2-figures/1-Guides/1.10.4_structure_quality_2.webp" alt="1.10.4_structure_quality_2" needTransformUrl="true" />
</div>

><p style="text-align:justify; line-height: 1.8; ">Independent of the 3D structure, AlphaFold produces an output called “Predicted Aligned Error” (PAE). This is shown at the bottom of structure pages as an interactive 2D plot.</p>
>
><p style="text-align:justify; line-height: 1.8; ">The colour at (x, y) indicates AlphaFold’s expected position error at residue x if the predicted and true structures were aligned on residue y. If the PAE is generally low for residue pairs x, y from two different domains, it indicates that AlphaFold predicts well-defined relative positions and orientations for them. If the PAE is generally high for residue pairs x, y from two different domains, then the relative positions and/or orientations of these domains in the 3D structure are uncertain and should not be interpreted.</p>
>
>---From [AlphaFoldDB](https://alphafold.ebi.ac.uk/)

<div style="text-align:center;">
    <img src="../data/2-figures/1-Guides/1.10.4_structure_quality_3.webp" alt="1.10.4_structure_quality_3" needTransformUrl="true" />
</div>

## Taxonomy

<div style="text-align:center;">
    <img src="../data/2-figures/1-Guides/1.10.5_taxonomy_1.webp" alt="1.10.5_taxonomy_1" needTransformUrl="true" />
</div>

Information about the species to which the receptor belongs.

* Species infomation
  * <p style="text-align:justify; line-height: 1.8; ">Scientific Name: The unique name for an organism such that two people can be sure that they are referring to the same organism.</p>
  * <p style="text-align:justify; line-height: 1.8; ">Taxon ID: The unique identifier assigned by the NCBI to the source organism</p>
  * Common Name: The species common name.
  * Other Names: The alternative name for this species.
  * Lineage: Class -> Order -> Family -> Genes
* Receptor infomation
  * \# of Functional ORs: Number of functional olfactory receptor genes in the species.
  * \# of Pseudogenes: Number of olfactory receptor pseudogenes in the species
  * Percentage of Pseudo: The percentage of olfactory receptor pseudogenes.
  * \# of Total ORs: Total number of functional olfactory receptor genes and pseudogenes in the species.
  * \# of Truncated ORs: Number of truncated genes in the species.
* Genome infomation
  * <p style="text-align:justify; line-height: 1.8; ">Assembly ID: The unique identifier for genomic assembly data in the NCBI database.</p>
  * <p style="text-align:justify; line-height: 1.8; ">Assembly Level: The highest level of genomic assembly for any object in the NCBI.</p>
  * <p style="text-align:justify; line-height: 1.8; ">Assembly Date: The date of submission of the genomic assembly data to the NCBI.</p>
  * <p style="text-align:justify; line-height: 1.8; ">Contig N50: Length such that sequence contigs of this length or longer include half the bases of the assembly.</p>
  * <p style="text-align:justify; line-height: 1.8; ">Genome Size: Genome size, measured in terms of the number of base pairs.</p>

<p style="text-align:justify; line-height: 1.8; ">In the upper right corner, an image representing the species to which the receptor belongs is displayed. For specific details regarding its source, please <a href="#/./2-DataIntroduction/2.3-DataDownload.md">refer to here</a>.</p>

<div style="text-align:center;">
    <img src="../data/2-figures/1-Guides/1.10.5_taxonomy_2.webp" alt="1.10.5_taxonomy_2" needTransformUrl="true" />
</div>
<p style="text-align:justify; line-height: 1.8; ">Bar chart depicting the receptor counts in various olfactory receptor subfamilies for this species, including functional olfactory receptors, olfactory receptor pseudogenes, and the total number of olfactory receptors. For details, please <a href="#/./2-DataIntroduction/2.6-CommunityDetection.md">refer to here</a>.</p>

## Sequence Similar Network

<div style="text-align:center;">
    <img src="../data/2-figures/1-Guides/1.10.6_network.webp" alt="1.10.6_network" needTransformUrl="true"/>
</div>
<span style="font-size:18px; color:red">①</span> Control button panel

* <p style="text-align:justify; line-height: 1.8; "><span style="font-size:16px; color:blue">ⓐ</span> Update Button: After adjusting the parameters, click this button to update the data.</p>
* <p style="text-align:justify; line-height: 1.8; "><span style="font-size:16px; color:blue">ⓑ</span> Help Button: Click to navigate to the BLAST documentation page.</p>

<span style="font-size:18px; color:red">②</span> Parameter panel.

* <p style="text-align:justify; line-height: 1.8; "><span style="font-size:16px; color:blue">ⓐ</span> Slider for adjusting edge weight threshold. Edges with values below this threshold will be filtered out.</p>
* <p style="text-align:justify; line-height: 1.8; "><span style="font-size:16px; color:blue">ⓑ</span> The scaling factor of the layout. The larger the value, the greater the repulsive force between the nodes.</p>
* <p style="text-align:justify; line-height: 1.8; "><span style="font-size:16px; color:blue">ⓒ</span> Stops the threshold of the layout and stops the layout when the global velocity factor of the layout is less than this threshold. Set to 0 to never stop.</p>
* <p style="text-align:justify; line-height: 1.8; "><span style="font-size:16px; color:blue">ⓓ</span> The gravity that the node is subjected to. This force will make the nodes close to the center.</p>
* <p style="text-align:justify; line-height: 1.8; "><span style="font-size:16px; color:blue">ⓔ</span> The weight distribution of the edges. Support is set to a single number, which is the uniform weight value.</p>
* <p style="text-align:justify; line-height: 1.8; "><span style="font-size:16px; color:blue">ⓕ</span> The number of iterations of an update. Because the force-guided algorithm usually draws the results of each iteration, but because the drawing time tends to be greater than the layout time, the layout efficiency will be reduced. At this time, we can set a larger steps parameter to ensure the layout and Draw time equalization and speed up the layout.</p>

<span style="font-size:18px; color:red">③</span> Pie chart illustrating the proportion of node categories.

<span style="font-size:18px; color:red">④</span> Histogram depicting the distribution of edge weights.

<span style="font-size:18px; color:red">⑤</span> Sequence Similar Network

* <span style="font-size:16px; color:blue">ⓐ</span> Presentation Space
* <span style="font-size:16px; color:blue">ⓑ</span> Download image button
* <span style="font-size:16px; color:blue">ⓒ</span> Number of nodes and edges

For details regarding the calculation of sequence similarity, please [refer to here](./2-DataIntroduction/2.9-SequenceSimilarity.md).


## Similar Proteins

<div style="text-align:center;">
    <img src="../data/2-figures/1-Guides/1.10.7_similar.webp" alt="1.10.7_similar" needTransformUrl="true"/>
</div>
<span style="font-size:18px; color:red">①</span> Selection button for choosing similar proteins at different threshold levels.

<span style="font-size:18px; color:red">②</span> Export and download buttons.

* <p style="text-align:justify; line-height: 1.8; "><span style="font-size:16px; color:blue">ⓐ</span> Export Table Button: Users can select the desired data entries using the checkboxes and then click this button to export the table data.</p>
* <p style="text-align:justify; line-height: 1.8; "><span style="font-size:16px; color:blue">ⓑ</span> Download Protein Sequences Button: Users can select the entries they wish to download protein sequences for by checking the corresponding boxes and then click the button to initiate the download.</p>

<span style="font-size:18px; color:red">③</span> Table of similar sequence information.

<span style="font-size:18px; color:red">④</span> Page adjustment panel.

* <p style="text-align:justify; line-height: 1.8; "><span style="font-size:16px; color:blue">ⓐ</span> The dropdown menu for setting the number of rows displayed per page offers five options: 10, 20, 30, 50, and 100. By default, 20 rows are displayed per page.</p>
* <p style="text-align:justify; line-height: 1.8; "><span style="font-size:16px; color:blue">ⓑ</span> Display the total number of results, along with buttons to navigate to specific pages.</p>

For detailed information on the clustering of similar sequences, please [refer to here](./2-DataIntroduction/2.10-CORDclust.md).


## Ligands

<div style="text-align:center;">
    <img src="../data/2-figures/1-Guides/1.10.8_ligands.webp" alt="1.10.8_ligands" needTransformUrl="true"/>
</div>
<p style="text-align:justify; line-height: 1.8; "><span style="font-size:18px; color:red">①</span> Green nodes represent olfactory receptors; clicking on them allows access to the detailed page of the receptor in the UniProt database. Blue nodes represent odor molecules that have elicited a response in experiments with the corresponding olfactory receptor; clicking on them allows access to the detailed page of the molecule in the PubChem database.</p>

<span style="font-size:18px; color:red">②</span> Original documents for data collection.

<span style="font-size:18px; color:red">③</span> The sources from which we obtain the data.

Most olfactory receptors lack experimentally validated ligands; therefore, the detailed pages for these receptors do not include a "Ligands" module. For detailed steps regarding data collection, please [refer to here](./2-DataIntroduction/2.14-Odorants.md).

<style>
table th:nth-of-type(1) {
    width: 10%;
}
table th:nth-of-type(2) {
    width: 20%;
}
table th:nth-of-type(3) {
    width: 10%;
}
</style>
