# 2.15 BLAST

<p style="text-align:justify; line-height: 1.8; ">The Basic Local Alignment Search Tool (BLAST) finds regions of local similarity between sequences. The program compares nucleotide or protein sequences to sequence databases and calculates the statistical significance of matches. BLAST can be used to infer functional and evolutionary relationships between sequences as well as help identify members of gene families.</p>

## Introduction

<p style="text-align:justify; line-height: 1.8; ">The BLAST functionality in CORD is implemented based on the NCBI BLAST+ toolkit (v2.15.0). We utilize the 'makeblastdb' sub-tool to construct BLAST sequence databases, including both protein and DNA sequences. The BLAST function for protein sequences is developed using the 'blastp' sub-tool, while the BLAST function for DNA sequences is developed using the 'blastn' sub-tool.</p>

## Constructing BLAST databases

```bash
# Constructing the BLAST Database for Protein Sequences
makeblastdb \
    -in Chordata_ORs_all_pros.fasta \
    -parse_seqids \
    -blastdb_version 5 \
    -taxid_map CORD_taxid_map.txt \
    -title CORD \
    -dbtype prot \
    -input_type fasta \
    -out ./CORD_blastpdb/CORD
```

```bash
# Constructing the BLAST Database for DNA Sequences
makeblastdb \
    -in Chordata_ORs_dnas.fasta \
    -parse_seqids \
    -blastdb_version 5 \
    -taxid_map CORD_taxid_map.txt \
    -title CORD \
    -dbtype nucl \
    -input_type fasta \
    -out ./CORD_blastpdbn/CORD
```

<p style="text-align:justify; line-height: 1.8; ">In the parameters, "Chordata_ORs_all_pros.fasta" stores all protein sequences in CORD in FASTA format. "Chordata_ORs_dnas.fasta" similarly stores all DNA sequences in CORD in FASTA format. The file "CORD_taxid_map.txt" records the mapping relationship between sequence names and corresponding species NCBI Taxonomy identifiers. The official explanations for the other parameters are provided below:</p>

* <p style="text-align:justify; line-height: 1.8; ">-in: Input file/database name</p>
* <p style="text-align:justify; line-height: 1.8; ">-parse_seqids: Option to parse seqid for FASTA input if set, for all other input types seqids are parsed automatically</p>
* <p style="text-align:justify; line-height: 1.8; ">-blastdb_version: Version of BLAST database to be created</p>
* <p style="text-align:justify; line-height: 1.8; ">-taxid_map: Text file mapping sequence IDs to taxonomy IDs.</p>
* <p style="text-align:justify; line-height: 1.8; ">-title: Title for BLAST database.</p>
* <p style="text-align:justify; line-height: 1.8; ">-dbtype: Molecule type of target db.  'prot' represents the protein database, while 'nucl' represents the nucleic acid database.</p>
* <p style="text-align:justify; line-height: 1.8; ">-input_type: Type of the data specified in input_file</p>
* <p style="text-align:justify; line-height: 1.8; ">-out: Name of BLAST database to be created</p>

For detailed information regarding the FASTA format, please [refer to this link](./2-DataIntroduction/2.17-FileFormat.md).

## BLAST command

```bash
# blastp command
blastp \
    -query $SeqName \
    -db $ProteinDB \
    -out $OutFile \
    -evalue $evalue \
    -matrix $ScoreMatrix \
    -word_size $wordSize \
    -max_target_seqs $maxTarget \
    -outfmt "6 sseqid slen qcovs bitscore evalue pident staxid" \
    -num_threads 6 \
    -taxids $taxid # Optional
```

```bash
# blastn command
blastn \
    -query $SeqName \
    -db $NuclDB \
    -out $OutFile \
    -evalue $evalue \
    -word_size $wordSize \
    -max_target_seqs $maxTarget \
    -outfmt "6 sseqid slen qcovs bitscore evalue pident staxid" \
    -num_threads 6 \
    -taxids $taxid # Optional
```

<p style="text-align:justify; line-height: 1.8; ">In the command line, the parameters '-query,' '-evalue,' '-matrix,' '-word_size,' 'max_target_seqs,' and '-taxids' are open interfaces for users to control. The remaining parameters are fixed on the backend. The official explanations for these parameters are as follows:</p>

* <p style="text-align:justify; line-height: 1.8; ">-query: Input file name.</p>
* <p style="text-align:justify; line-height: 1.8; ">-db: BLAST database name.</p>
* <p style="text-align:justify; line-height: 1.8; ">-out: Output file name.</p>
* <p style="text-align:justify; line-height: 1.8; ">-evalue: Expectation value (E) threshold for saving hits.</p>
* <p style="text-align:justify; line-height: 1.8; ">-matrix: Scoring matrix name (normally BLOSUM62). For detailed information about the BLOSUM62 matrix, please <a href="#/./2-DataIntroduction/2.19-BLOSUMmatrix.md">refer to here</a>.</p>
* <p style="text-align:justify; line-height: 1.8; ">-word_size: Word size for wordfinder algorithm.</p>
* <p style="text-align:justify; line-height: 1.8; ">-max_target_seqs: Maximum number of aligned sequences to keep.</p>
* <p style="text-align:justify; line-height: 1.8; ">-outfmt: Output file format.</p>
* <p style="text-align:justify; line-height: 1.8; ">-num_threads: Number of threads (CPUs) to use in the BLAST search</p>
* <p style="text-align:justify; line-height: 1.8; ">-taxids: Restrict search of database to include only the specified taxonomy IDs and their descendants (multiple IDs delimited by ',')</p>

## Reference

<p style="text-align:justify; line-height: 1.8; ">Camacho C, Coulouris G, Avagyan V, Ma N, Papadopoulos J, Bealer K, Madden TL. BLAST+: architecture and applications. <i>BMC Bioinformatics</i>. 2009 Dec 15;10:421.</p>

