# 2.17 File Format

<p style="text-align:justify; line-height: 1.8; ">In CORD, various file formats are employed. Here, we choose to introduce several commonly used file formats.</p>

## FASTA Format

<p style="text-align:justify; line-height: 1.8; ">In bioinformatics, the FASTA format is a text-based format for representing either nucleotide sequences or amino acid (protein) sequences, in which nucleotides or amino acids are represented using single-letter codes.</p>

<b>Overview</b>

<p style="text-align:justify; line-height: 1.8; ">A sequence in FASTA format begins with a single-line description, followed by lines of sequence data. The description line is distinguished from the sequence data by a greater-than (">") symbol in the first column. The word following the ">" symbol is the identifier of the sequence, and the rest of the line is the description (both are optional). There should be no space between the ">" and the first letter of the identifier. It is recommended that all lines of text be shorter than 80 characters. The sequence ends if another line starting with a ">" appears; this indicates the start of another sequence.</p>

<b>Example</b>

<textarea rows="8" style="resize:none; width:100%; font-family: Courier, monospace, Menlo, Monaco, Courier New;" disabled>
>sp|Q8NGI8|O5AN1_HUMAN Olfactory receptor 5AN1 OS=Homo sapiens OX=9606 GN=OR5AN1 PE=2 SV=1
MTGGGNITEITYFILLGFSDFPRIIKVLFTIFLVIYITSLAWNLSLIVLIRMDSHLHTPM
YFFLSNLSFIDVCYISSTVPKMLSNLLQEQQTITFVGCIIQYFIFSTMGLSESCLMTAMA
YDRYAAICNPLLYSSIMSPTLCVWMVLGAYMTGLTASLFQIGALLQLHFCGSNVIRHFFC
DMPQLLILSCTDTFFVQVMTAILTMFFGIASALVIMISYGYIGISIMKITSAKGRSKAFN
TCASHLTAVSLFYTSGIFVYLSSSSGGSSSFDRFASVFYTVVIPMLNPLIYSLRNKEIKD
ALKRLQKRKCC
</textarea>

The nucleic acid codes are:

<table style="width: 50%;">
  <tr>
    <td>A --> adenosine</td>
    <td>M --> A C (amino)</td>
  </tr>
  <tr>
    <td>C --> cytidine</td>
    <td>S --> G C (strong)</td>
  </tr>
  <tr>
    <td>G --> guanine</td>
    <td>W --> A T (weak)</td>
  </tr>
  <tr>
    <td>T --> thymidine</td>
    <td>B --> G T C</td>
  </tr>
  <tr>
    <td>U --> uridine</td>
    <td>D --> G A T</td>
  </tr>
  <tr>
    <td>R --> G A (purine)</td>
    <td>H --> A C T</td>
  </tr>
  <tr>
    <td>Y --> T C (pyrimidine)</td>
    <td>V --> G C A</td>
  </tr>
  <tr>
    <td>K --> G T (keto)</td>
    <td>N --> A G C T (any)</td>
  </tr>
</table>


The accepted amino acid codes are:

| Amino acid | Three letter code | One letter code             |
| ---------- | ----------------- | --------------------------- |
| A          | ALA               | alanine                     |
| C          | CYS               | cystine                     |
| D          | ASP               | aspartate                   |
| E          | GLU               | glutamate                   |
| F          | PHE               | phenylalanine               |
| G          | GLY               | glycine                     |
| H          | HIS               | histidine                   |
| I          | ILE               | isoleucine                  |
| K          | LYS               | lysine                      |
| L          | LEU               | leucine                     |
| M          | MET               | methionine                  |
| N          | ASN               | asparagine                  |
| P          | PRO               | proline                     |
| Q          | GLN               | glutamine                   |
| R          | ARG               | arginine                    |
| S          | SER               | serine                      |
| T          | THR               | threonine                   |
| V          | VAL               | valine                      |
| W          | TRP               | tryptophan                  |
| Y          | TYR               | tyrosine                    |
| B          | ASX               | aspartate or asparagine     |
| Z          | GLX               | glutamate or glutamine      |
| X          | <p></p>           | any                         |
| U          | <p></p>           | selenocysteine              |
| *          | <p></p>           | translation stop            |
| -          | <p></p>           | gap of indeterminate length |

## Sequence alignment formats

<b>1. FASTA format</b>

> <p style="text-align:justify; line-height: 1.8; ">The FASTA format consists of aligned fasta, in which lower and upper case are equivalent; ' . ' and ' - ' are equivalent.</p>
>
> --- From [trRosetta](https://yanglab.qd.sdu.edu.cn/trRosetta/msa_format.html)

<b>Example</b>

<textarea rows="15" style="resize:none; width:100%; font-family: Courier, monospace, Menlo, Monaco, Courier New;" disabled>
>Q8NGI8
MTGGGNITEITYFILLGFSDFPRIIKVLFTIFLVIYITSLAWNLSLIVLIRMDSHLHTPM
YFFLSNLSFIDVCYISSTVPKMLSNLLQEQQTITFVGCIIQYFIFSTMGLSESCLMTAMA
YDRYAAICNPLLYSSIMSPTLCVWMVLGAYMTGLTASLFQIGALLQLHFCGSNVIRHFFC
DMPQLLILSCTDTFFVQVMTAILTMFFGIASALVIMISYGYIGISIMKITSAKGRSKAFN
TCASHLTAVSLFYTSGIFVYLSSSSGGSSSFDRFASVFYTVVIPMLNPLIYSLRNKEIKD
ALKRLQKRKCC---
>P34982
-MDGGNQSEGSEFLLLGMSESPEQQRILFWMFLSMYLVTVVGNVLIILAISSDSRLHTPV
YFFLANLSFTDLFFVTNTIPKMLVNLQSHNKAISYAGCLTQLYFLVSLVALDNLILAVMA
YDRYVAICCPLHYTTAMSPKLCILLLSLCWVLSVLYGLIHTLLMTRVTFCGSRKIHYIFC
EMYVLLRMACSNIQINHTVLIATGCFIFLIPFGFVIISYVLIIRAILRIPSVSKKYKAFS
TCASHLGAVSLFYGTLCMVYLKPLHT-YSVKDSVATVMYAVVTPMMNPFIYSLRNKDMHG
ALGRLLDKHFKRLT
</textarea>

<b>2. A3M format</b>

> <p style="text-align:justify; line-height: 1.8; ">The A3M format consists of aligned fasta, in which alignments are shown with inserts as lower case characters, matches as upper case characters, deletions as ' - ', and gaps aligned to inserts as ' . '. Note that gaps aligned to inserts can be omitted in the A3M format.</p>
>
> --- From [trRosetta](https://yanglab.qd.sdu.edu.cn/trRosetta/msa_format.html)

<b>Example</b>

<textarea rows="15" style="resize:none; width:100%; font-family: Courier, monospace, Menlo, Monaco, Courier New;" disabled>
>A002S3F
MANWNNTYFTEFILLGLTDNPKLEVILFILFLIFYIITLTGNLGILVAIRADSRLHTPMY
FFLNNLSFLDLCYATIITPKTLVNFMSKSKTIGYKECAVQMYFFAASVTTECFLLGIMAY
DRYVAICNPLLYSVVMNKKICIQLVAGAYTLGYLNATLHTITTFRLPFCKTNKIDHFYCD
VPPLLKLSCTKTTMNEILMFIFGGFAETSSLTTIIVSYTYIITSILRMRSADGRKKAFST
CGSHFTAVTIFYSTILFMYLRPASTYSMSQDRVASVFYSVIIPMLNPLIYSLRNNDVAQA
LKKIRTKYCCKGK
>AGP04LF
MA--NQTILSEFFLSGLSDLPSLQLPLFLIFLLIYLLTLTWNLLIISLIVTDSHLHNPMY
FFLGNLACLDLCYSSVTIPRMLFDLHTKRRVITIPACITQVFLFFFFAGYEVFLLAIMSI
DRYLAICRPLHYMQIMRWRVCVQLALGVWSLGFIHSLLHTLYSLKLTFCGSNTIKSFFCD
LPKLFQISCSDVFINVLLIFLVGGFLAVGSLMMTFLPYVFIIKTVLKIQVEGNRKKVFST
CSSHLTVVFIFYAACMFNYFRPSSGNQFTSDIVMSVFYTVITPLLNPLIYSLRNQDLRSA
FgailmrtasgvlKKIV--------
</textarea>

## PDB Format

<b>Overview</b>

<p style="text-align:justify; line-height: 1.8; ">The Protein Data Bank (PDB) is an archive of experimentally determined three-dimensional structures of biological macromolecules that serves a global community of researchers, educators, and students. The data contained in the archive include atomic coordinates, crystallographic structure factors and NMR experimental data. Aside from coordinates, each deposition also includes the names of molecules, primary and secondary structure information, sequence database references, where appropriate, and ligand and biological assembly information, details about data collection and structure solution, and bibliographic citations.</p>

<p style="text-align:justify; line-height: 1.8; ">In the PDB file format, there are numerous sections. Here, we will only introduce the essential contents of the Coordinate Section, specifically the ATOM entries.</p>

| COLUMNS | DATA  TYPE   | FIELD      | DEFINITION                                 |
| ------- | ------------ | ---------- | ------------------------------------------ |
| 1 -  6  | Record name  | "ATOM  "   |                                            |
| 7 - 11  | Integer      | serial     | Atom serial number.                        |
| 13 - 16 | Atom         | name       | Atom name.                                 |
| 17      | Character    | altLoc     | Alternate location indicator.              |
| 18 - 20 | Residue name | resName    | Residue name.                              |
| 22      | Character    | chainID    | Chain identifier.                          |
| 23 - 26 | Integer      | resSeq     | Residue sequence number.                   |
| 27      | AChar        | iCode      | Code for insertion of residues.            |
| 31 - 38 | Real(8.3)    | x          | Orthogonal coordinates for X in Angstroms. |
| 39 - 46 | Real(8.3)    | y          | Orthogonal coordinates for Y in Angstroms. |
| 47 - 54 | Real(8.3)    | z          | Orthogonal coordinates for Z in Angstroms. |
| 55 - 60 | Real(6.2)    | occupancy  | Occupancy.                                 |
| 61 - 66 | Real(6.2)    | tempFactor | Temperature  factor.                       |
| 77 - 78 | LString(2)   | element    | Element symbol, right-justified.           |
| 79 - 80 | LString(2)   | charge     | Charge  on the atom.                       |

<b>Example</b>

<textarea rows="26" style="resize:none; width:100%; font-family: Courier, monospace, Menlo, Monaco, Courier New;" disabled>
ATOM      1  N   CYS A   4     122.936 121.888  60.724  1.00 86.79           N  
ATOM      2  CA  CYS A   4     122.270 122.455  59.557  1.00 86.79           C  
ATOM      3  C   CYS A   4     122.594 123.938  59.395  1.00 86.79           C  
ATOM      4  O   CYS A   4     123.595 124.422  59.923  1.00 86.79           O  
ATOM      5  CB  CYS A   4     120.759 122.241  59.653  1.00 86.79           C  
ATOM      6  SG  CYS A   4     120.015 122.896  61.155  1.00 86.79           S  
ATOM      7  N   ASN A   5     121.742 124.656  58.658  1.00 88.49           N  
ATOM      8  CA  ASN A   5     122.058 126.031  58.285  1.00 88.49           C  
ATOM      9  C   ASN A   5     121.859 127.019  59.431  1.00 88.49           C  
ATOM     10  O   ASN A   5     122.581 128.021  59.502  1.00 88.49           O  
ATOM     11  CB  ASN A   5     121.226 126.451  57.070  1.00 88.49           C  
ATOM     12  CG  ASN A   5     119.740 126.505  57.364  1.00 88.49           C  
ATOM     13  OD1 ASN A   5     119.109 125.482  57.632  1.00 88.49           O  
ATOM     14  ND2 ASN A   5     119.170 127.704  57.313  1.00 88.49           N  
ATOM     15  N   PHE A   6     120.908 126.770  60.329  1.00 84.90           N  
ATOM     16  CA  PHE A   6     120.639 127.710  61.408  1.00 84.90           C  
ATOM     17  C   PHE A   6     120.105 126.960  62.619  1.00 84.90           C  
ATOM     18  O   PHE A   6     119.602 125.839  62.510  1.00 84.90           O  
ATOM     19  CB  PHE A   6     119.655 128.805  60.970  1.00 84.90           C  
ATOM     20  CG  PHE A   6     118.207 128.418  61.104  1.00 84.90           C  
ATOM     21  CD1 PHE A   6     117.646 127.463  60.268  1.00 84.90           C  
ATOM     22  CD2 PHE A   6     117.402 129.024  62.059  1.00 84.90           C  
ATOM     23  CE1 PHE A   6     116.312 127.111  60.389  1.00 84.90           C  
ATOM     24  CE2 PHE A   6     116.068 128.677  62.185  1.00 84.90           C  
ATOM     25  CZ  PHE A   6     115.523 127.719  61.349  1.00 84.90           C  
</textarea>

## CD-HIT clustering format

<b>Overview</b>

<p style="text-align:justify; line-height: 1.8; ">The <a target="_blank" href="https://sites.google.com/view/cd-hit">CD-HIT</a> clustering file contains the clustering results, usually with the file extension ".clstr". Each cluster begins with a ">Cluster 1" header line, where the ">Cluster" symbol is the identifier for the header line, and the number "1" is the cluster identifier. This identifier increases sequentially from the top to the bottom of the file. Following the header line, there are several lines containing information about a representative sequence and several other member sequences belonging to the cluster. Each line of sequence information consists of four main parts: the first part is an internal cluster identifier, the second part is the sequence length, the third part is the sequence name, and the fourth part is the sequence similarity to the reference sequence (if this position is "*" it indicates that the sequence is the representative sequence).</p>

<b>Example</b>

<textarea rows="11" style="resize:none; width:100%; font-family: Courier, monospace, Menlo, Monaco, Courier New;" disabled>
>Cluster 1
0   500aa, >A0A5N4DLY7... *
1   314aa, >MHBRKLF... at 96.98%
2   314aa, >MOAJFYF... at 96.17%
3   314aa, >M0W12AF... at 99.49%
>Cluster 2
0   494aa, >A0A5N4CL90... *
1   312aa, >M7H50YF... at 95.96%
2   312aa, >MTI3N5F... at 98.99%
3   312aa, >MTSGPYF... at 95.52%
</textarea>

## EdgeList format

<b>Overview</b>

<p style="text-align:justify; line-height: 1.8; ">The EdgeList file is a representation format for graph data, commonly used to describe the edge relationships in a graph. It is a text file where each line represents an edge in the graph. In CORD, we utilize the EdgeList format to store data for sequence similarity networks.</p>

<b>Example</b>

<textarea rows="8" style="resize:none; width:100%; font-family: Courier, monospace, Menlo, Monaco, Courier New;" disabled>
Node1 Node2 Weight
1     2     0.5
1     3     0.9
1     4     0.4
2     4     0.7
3     4     0.8
3     5     0.6
</textarea>

<p style="text-align:justify; line-height: 1.8; ">Node1 and Node2 are identifiers for two nodes in the graph, used in CORD to represent the names of sequences. Weight represents the weight of the edge connecting two nodes, utilized in CORD to indicate the similarity between sequences, including sequence identity, sequence similarity, and sequence similarity score. Regarding the description of sequence similarity, please <a href="#/./2-DataIntroduction/2.9-SequenceSimilarity.md">refer to this section</a>.</p>

## Tabular formats

<p style="text-align:justify; line-height: 1.8; ">We utilize four formats—CSV, TSV, Excel, and JSON—to store and transmit tabular data.</p>

1. <p style="text-align:justify; line-height: 1.8; ">CSV (Comma-Separated Values):<br/>
      Features: Data fields are separated by commas, with each line representing a record.<br/>
      Advantages: Simple, human-readable, widely supported.<br/>
      Use Cases: Suitable for simple data structures, easy to process, and import/export.
   </p>
2. <p style="text-align:justify; line-height: 1.8; ">TSV (Tab-Separated Values):<br/>
       Features: Similar to CSV but uses tabs instead of commas for field separation.<br/>
   	Advantages: Suitable for text data containing commas, avoids conflicts with commas in CSV.<br/>
   	Use Cases: Similar to CSV, used for processing text data containing commas.
   </p>

3. <p style="text-align:justify; line-height: 1.8; ">Excel (XLSX):<br/>
       Features: Uses spreadsheet format, supports multiple sheets, can include formulas, charts, etc.<br/>
       Advantages: Powerful, supports large datasets, and complex calculations.<br/>
       Use Cases: Widely used for business and scientific data analysis, suitable for complex computations and visualizations.
   </p>
4. <p style="text-align:justify; line-height: 1.8; ">JSON (JavaScript Object Notation):<br/>
       Features: Data is represented using key-value pairs with a hierarchical structure, easy to parse and generate.<br/>
       Advantages: Supports complex data structures, nesting, suitable for web applications and APIs.<br/>
       Use Cases: Used for storing and exchanging structured data, commonly used in web development.
   </p>

## Reference

<p style="text-align:justify; line-height: 1.8; ">Berman H, Henrick K, Nakamura H. Announcing the worldwide Protein Data Bank. <i>Nat Struct Biol</i>. 2003;10(12):980.</p>

<p style="text-align:justify; line-height: 1.8; ">Li W, Godzik A. Cd-hit: a fast program for clustering and comparing large sets of protein or nucleotide sequences. <i>Bioinformatics</i>. 2006;22(13):1658-1659.</p>

<style>
table th:nth-of-type(1) {
    width: 10%;
}
table th:nth-of-type(2) {
    width: 10%;
}
table th:nth-of-type(3) {
    width: 10%;
}
table th:nth-of-type(4) {
    width: 30%;
}
</style>
