# FAQs

<p style="text-align:justify; line-height: 1.8; ">1. How to search for genes/proteins?<br/>
You can enter the gene ID or protein name in the search box on the homepage, then press the Enter key on the keyboard or click the 'Search' button on the right end of the search box using the mouse to initiate the search. On pages other than the homepage, you can find the search box in the top right corner of the page, and its usage is the same as the search box on the homepage.</p>
<p style="text-align:justify; line-height: 1.8; ">2. What content can I search for in the database?<br/>
In CORD, you can search for various types of content, including protein identifiers (CORD ID), gene names, UniProt identifiers, gene families, gene subfamilies, gene clades, NCBI assembly IDs, NCBI taxon IDs, scientific names of species, common names of species, lineages (genus, family, order, class), odor CAS, odor type, odor strength, and more.</p>
<p style="text-align:justify; line-height: 1.8; ">3. How to download specific data or information? <br/>
In this database, there are various types of data available for download. Generally, you can use the download button (a downward arrow, usually located in the top right corner) to obtain downloads of specific data, including protein sequences, DNA sequences, images, protein structures, and more. You can also use the export button to obtain tabular data, including search result lists, BLAST results, and more.</p>
<p style="text-align:justify; line-height: 1.8; ">4. Is the data in the CORD database free?<br/>
Certainly, you are free to use all data in this database for academic purposes. If you find our data helpful for your research, we would greatly appreciate it if you could <a href="#/./5-CiteUs.md">cite our work</a>.</p>


<p style="text-align:justify; line-height: 1.8; ">5. What analysis tools are available?<br/>
You can use analysis tools such as global sequence similarity networks, PSSM, Weblogo, community analysis, BLAST, Snake Diagram, protein structure viewer, and more.</p>

<p style="text-align:justify; line-height: 1.8; ">6. How frequently is the database updated?<br/>
Based on previous experience, we typically perform data updates approximately every six months, with a major database-wide update every two years.</p>

<p style="text-align:justify; line-height: 1.8; ">7. How to determine which version of the database I am using?<br/>
In CORD, each data entry is timestamped, providing a record of when it was annotated. This timestamp allows you to ascertain the version of the data.</p>

<p style="text-align:justify; line-height: 1.8; ">8. Can I use CORD to discover homologous sequences of olfactory receptors that I am interested in?<br/>
Certainly. You can achieve this by utilizing our provided BLAST functionality, which includes both protein and DNA sequences.</p>

<p style="text-align:justify; line-height: 1.8; ">9. Can I cite the database in my research?<br/>
Certainly, we would be delighted if our data can be of assistance to you.</p>

<p style="text-align:justify; line-height: 1.8; ">10. How do you assess the quality of data in the CORD?<br/>
The core data in CORD is derived from the annotation results of genome assembly. There are two aspects to evaluate the quality of the data: On one hand, metrics such as Contig N50 and assembly level indicators reflect the quality of genome assembly. On the other hand, annotation quality is indicated by the 'Pattern Match Score.</p>

<p style="text-align:justify; line-height: 1.8; ">11. How do I annotate the olfactory receptor genes in the genome if I have sequenced and assembled the genome of a chordate myself?<br/>
We have developed a tool for annotating olfactory receptor genes from chordate genomes, and you can download and install it from <a target="_blank" href="https://github.com/ToHanwei/Genome2OR">GitHub</a> for use.</p>

<p style="text-align:justify; line-height: 1.8; ">12. What is the source of the data?<br/>
We obtained complete genome assembly data for all chordates from the NCBI Assembly database (as of March 2023). We annotated these genomes and derived the core data of CORD, which includes olfactory receptor genes of chordates. Of course, CORD encompasses various types of data, with the majority being analyzed and computed by us. Additionally, some data are sourced from databases such as UniProt, HORDE, ORDB, OlfactionBase, and others.</p>

<p style="text-align:justify; line-height: 1.8; ">13. How does the database protect user privacy?<br/>
CORD does not require registration or login, and it does not record your personal information during your usage.</p>

<p style="text-align:justify; line-height: 1.8; ">14. Are there recommended best practices for using the database?<br/>
You can find the details on how to use CORD in the Guides section on the Help page.</p>

<p style="text-align:justify; line-height: 1.8; ">15. How many sequences can I submit for a BLAST search at one time?<br/>
The BLAST functionality in CORD allows the input of only one query sequence at a time.</p>